File: AoC2016/JavaScript/day1-2.js
//puzzle setup
const fs = require('fs');
var input = fs.readFileSync('../day1in.txt','utf8');
input = input.split(', ');

//do stuff
var coords = [[0,0]];
var curr = [...coords[0]];
var facing = 0; //0 = north, 1 = east, 2 = south, 3 = west
for (i in 8) {console.log(i)}
for (let i = 0; i < input.length; i++)
{
    newFacing(input[i]);
    let steps = Number(input[i].substring(1));
    let j = 0;
    switch(facing)
    {
        case 0:
            for (; j < steps; j++)
            {
                curr[1]++;
                beenThere(curr);
                coords.push([...curr]);
            }
            break;
        case 1:
            for (; j < steps; j++)
            {
                curr[0]++;
                beenThere(curr);
                coords.push([...curr]);
            }
            break;
        case 2:
            for (; j < steps; j++)
            {
                curr[1]--;
                beenThere(curr);
                coords.push([...curr]);
            }
            break;
        case 3:
            for (; j < steps; j++)
            {
                curr[0]--;
                beenThere(curr);
                coords.push([...curr]);
            }
            break;
    }
}

function newFacing(n)
{
    let change = n.substring(0,1);
    if (change === 'R')
    {
        facing < 3 ? facing++ : facing = 0;
    } else {
        facing > 0 ? facing-- : facing = 3;
    }
}

function beenThere(loc)
{
    let check = -1;
    for (i in coords)
    {
        if (coords[i][0] == loc[0] && coords[i][1] == loc[1]) check = i;
    }
    if (check > -1)
    {
        console.log(coords[check].reduce((a,b)=>Math.abs(a)+Math.abs(b)));
        process.exit(1);
    }
}

File: AoC2016/Python3/day1-2.py
#puzzle setup
with open('./day1in.txt','r') as pInput:
    pInput = pInput.read()

#do stuff
pInput = pInput.split(', ')
coords = [(0,0)]
facing = 0

def stepCheck(coord):
    if (len(coords) - coords[::-1].index(coord) - 1 != coords.index(coord)):
        print(abs(coord[0]) + abs(coord[1]))
        quit()

for i in range(0,len(pInput)):
    facing = ((facing - 1) % 4 if pInput[i][:1] == 'L' else (facing + 1) % 4)
    move = int(pInput[i][1:])
    if (facing == 0):
        for j in range(0,move):
            coords.append( (coords[-1][0], coords[-1][1] + 1) ) #north
            stepCheck(coords[-1])
    elif (facing == 1):
        for j in range(0,move):
            coords.append( (coords[-1][0] + 1, coords[-1][1]) ) #east
            stepCheck(coords[-1])
    elif (facing == 2):
        for j in range(0,move):
            coords.append( (coords[-1][0], coords[-1][1] - 1) ) #south
            stepCheck(coords[-1])
    else:
        for j in range(0,move):
            coords.append( (coords[-1][0] - 1, coords[-1][1]) ) #west
            stepCheck(coords[-1])
